src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md

```diff
@@ -0,0 +1,636 @@
+      // See the [lazy properites](/web/fundamentals/architecture/building-components/best-practices#lazy-properties)
+      // section for more details.
+      this._upgradeProperty('checked');
+      this._upgradeProperty('disabled');
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
+
+    _upgradeProperty(prop) {
+      if (this.hasOwnProperty(prop)) {
+        let value = this[prop];
+        delete this[prop];
+        this[prop] = value;
+      }
+    }
+
+    /**
+     * `disconnectedCallback()` fires when the element is removed from the DOM.
+     * It's a good place to do clean up work like releasing references and
+     * removing event listeners.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('click', this._onClick);
+    }
+
+    /**
+     * Properties and their corresponding attributes should mirror one another.
+     * The property setter for `checked` handles truthy/falsy values and
+     * reflects those to the state of the attribute. See the [avoid
+     * reentrancy](/web/fundamentals/architecture/building-components/best-practices#avoid-reentrancy)
+     * section for more details.
+     */
+    set checked(value) {
+      const isChecked = Boolean(value);
+      if (isChecked)
+        this.setAttribute('checked', '');
+      else
+        this.removeAttribute('checked');
+    }
+
+    get checked() {
+      return this.hasAttribute('checked');
+    }
+
+    set disabled(value) {
+      const isDisabled = Boolean(value);
+      if (isDisabled)
+        this.setAttribute('disabled', '');
+      else
+        this.removeAttribute('disabled');
+    }
+
+    get disabled() {
+      return this.hasAttribute('disabled');
+    }
+
+    /**
+     * `attributeChangedCallback()` is called when any of the attributes in the
+     * `observedAttributes` array are changed. It's a good place to handle
+     * side effects, like setting ARIA attributes.
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+      const hasValue = newValue !== null;
+      switch (name) {
+        case 'checked':
+          this.setAttribute('aria-checked', hasValue);
+          break;
+        case 'disabled':
+          this.setAttribute('aria-disabled', hasValue);
+          // The `tabindex` attribute does not provide a way to fully remove
+          // focusability from an element.
+          // Elements with `tabindex=-1` can still be focused with
+          // a mouse or by calling `focus()`.
+          // To make sure an element is disabled and not focusable, remove the
+          // `tabindex` attribute.
+          if (hasValue) {
+            this.removeAttribute('tabindex');
+            // If the focus is currently on this element, unfocus it by
+            // calling the `HTMLElement.blur()` method.
+            if (document.activeElement === this)
```

Comment:
This check won't work if `howto-checkbox` is contained in a shadowRoot (`document.activeElement` will be the host of that shadowRoot)
Is it bad to just invoke `this.blur()`?